runtime/Go/antlr/atn_simulator.go

```diff
@@ -25,3 +32,15 @@ func (b *BaseATNSimulator) getCachedContext(context PredictionContext) Predictio
 
 	return getCachedBasePredictionContext(context, b.sharedContextCache, visited)
 }
+
+func (b *BaseATNSimulator) SharedContextCache() *PredictionContextCache {
+	return b.sharedContextCache
+}
+
+func (b *BaseATNSimulator) ATN() *ATN {
+	return b.atn
+}
+
+func (b *BaseATNSimulator) DecisionToDFA() []*DFA {
+	return b.decisionToDFA
+}
```

Comment:
Note, I still don't like the `IFoo` style of naming interfaces, but there's not really a good way to do this. It might be that we move to `ATNSimulator` for the interface and `ATNSimulatorImpl` for the implementation. This seems more idiomatic, I suppose.